crates/pop-parachains/src/build.rs

```diff
@@ -75,22 +75,29 @@ pub fn binary_path(target_path: &Path, node_path: &Path) -> Result<PathBuf, Erro
 /// * `binary_path` - The path to the node binary executable that contains the `build-spec` command.
 /// * `plain_chain_spec` - Location of the plain_parachain_spec file to be generated.
 /// * `default_bootnode` - Whether to include localhost as a bootnode.
+/// * `chain` - The chain specification. It can be one of the predefined ones (e.g. dev, local or a
+/// 	custom one) or the path to an existing chain spec.
 pub fn generate_plain_chain_spec(
 	binary_path: &Path,
 	plain_chain_spec: &Path,
 	default_bootnode: bool,
-	chain: Option<&str>,
+	chain: &str,
 ) -> Result<(), Error> {
 	check_command_exists(binary_path, "build-spec")?;
 	let mut args = vec!["build-spec"];
-	if let Some(chain_spec_id) = chain {
-		args.push("--chain");
-		args.push(chain_spec_id);
-	}
+	args.push("--chain");
+	args.push(chain);
 	if !default_bootnode {
 		args.push("--disable-default-bootnode");
 	}
-	cmd(binary_path, args).stdout_path(plain_chain_spec).stderr_null().run()?;
+	// Create a temporary file.
+	let temp_file = tempfile::NamedTempFile::new_in(std::env::temp_dir())?;
+	// Run the command and redirect output to the temporary file.
+	cmd(binary_path, args).stdout_path(temp_file.path()).stderr_null().run()?;
+	// Atomically replace the chain spec file with the temporary file.
+	let _ = temp_file
+		.persist(plain_chain_spec)
+		.map_err(|_| Error::AnyhowError(anyhow::anyhow!("error")));
```

Comment:
This change was necessary because previously it would empty the chain spec file before executing the command, and would thus fail.